client/Hooks/useLogger/Hook.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import { useEffect, useRef } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+export const MESSAGE_BUFFER_MAX_SIZE = 100;
+// TODO: Replace with the regex from the LOGGING query param
+const defaultLoggingRegex = /.+/;
+// TODO: Replace with the web socket address from config
+const defaultWebSocketAddress = 'ws://localhost:3000/log';
+
+// Unique client ID created when this module is loaded
+const clientID = uuidv4();
+
+export type LogLevelType =
+  | 'fatal'
+  | 'error'
+  | 'warn'
+  | 'info'
+  | 'debug'
+  | 'trace';
+
+interface loggerMessage {
+  clientTime: number;
+  clientID: string;
+  clientLevel: LogLevelType;
+  componentName: string;
+  msg: string;
+}
+
+export type LoggerType = {
+  log: (clientLevel: LogLevelType, msg: string) => void;
+  fatal: (msg: string) => void;
+  error: (msg: string) => void;
+  warn: (msg: string) => void;
+  info: (msg: string) => void;
+  debug: (msg: string) => void;
+  trace: (msg: string) => void;
+};
+
+export const useLogger: (componentName: string) => LoggerType = (
+  componentName: string
+) => {
+  // TODO: store the logger websocket in context, so you get one for the whole UI, rather than one every time you call useLogger.
+  const loggerWebSocketRef = useRef<WebSocket | null>(null);
+  const loggerMessageBufferRef = useRef<loggerMessage[]>([]);
+
+  useEffect(() => {
+    // Check if the component should be logged
+    if (defaultLoggingRegex.test(componentName)) {
+      const loggerWebSocket = loggerWebSocketRef.current;
```

Comment:
Did we want to declare these types instead of using inference?

Reply:
This code has been updated, hopefully the types are clearer now.